src/components/AddItemButton.tsx

```diff
@@ -0,0 +1,68 @@
+'use client'
+
+import { addItemAction } from '@/app/actions'
+import { formatErrorMessage } from '@/lib/utils'
+import { Add } from '@mui/icons-material'
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Stack,
+  TextField,
+} from '@mui/material'
+import { useSnackbar } from 'notistack'
+import { useState } from 'react'
+
+export const AddItemButton = () => {
+  const { enqueueSnackbar } = useSnackbar()
+  const [open, setOpen] = useState(false)
+
+  const handleOpen = () => setOpen(true)
+  const handleClose = () => {
+    setOpen(false)
+  }
+
+  const handleSubmit = async (formData: FormData) => {
+    try {
+      await addItemAction(formData)
```

Comment:
Note that even though we this is a client-rendered component, we can still call server actions from it (i.e. `addItemAction`) -- cool!